File: 1-two-sum/1-two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    hash = Hash.new
    arr = []
    
    nums.each_with_index do |num, index|
        number = target - num
        if hash.include?(number)
            arr << index
            arr << hash[number]
        else    
            hash[num] = index
        end
    end
    
    arr
end

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> m;
      
        for(int i{0}; i < nums.size(); ++i){
          auto it = m.find(target - nums.at(i));
          
          if ( it != m.end()){
            return std::vector<int> {it->second, i};
          }else {
            m.insert({nums.at(i), i});
          }
        }
      return std::vector<int> {-1};
    }
};